pkg/agent/flowexporter/connections/conntrack_linux_test.go

```diff
@@ -101,19 +113,85 @@ func TestConnTrackSystem_DumpFlows(t *testing.T) {
 		GatewayConfig: gwConfig,
 		PodIPv4CIDR:   podCIDR,
 	}
-	// Test the DumpFlows implementation of connTrackSystem
-	mockNetlinkCT := connectionstest.NewMockNetFilterConnTrack(ctrl)
-	connDumperDPSystem := NewConnTrackSystem(nodeConfig, svcCIDR, netip.Prefix{}, false)
+	testCases := []struct {
+		name                string
+		protocols           []string
+		testFlows           []*flowexporter.Connection
+		expectedConnections int
+	}{
+		{
+			"Filter unhandled flow types",
+			nil,
+			testVarietyFlows,
+			1,
+		},
+		{
+			"Filter for all supported protocols with explicit declaration",
+			[]string{"tcp", "udp", "sctp"},
+			testFlowsMixedProtocols,
+			6,
+		},
+		{
+			"Filter for all supported protocols with default config",
+			nil,
+			testFlowsMixedProtocols,
+			6,
+		},
+		{
+			"Filter for only TCP flows",
+			[]string{"tcp"},
+			testFlowsMixedProtocols,
+			3,
+		},
+		{
+			"Filter for only UDP",
+			[]string{"udp"},
+			testFlowsMixedProtocols,
+			2,
+		},
+		{
+			"Filter for only SCTP",
+			[]string{"sctp"},
+			testFlowsMixedProtocols,
+			1,
+		},
+		{
+			"Filter for only TCP and SCTP",
+			[]string{"tcp", "sctp"},
+			testFlowsMixedProtocols,
+			4,
+		},
+		{
+			"Filter for UDP and SCTP",
+			[]string{"udp", "sctp"},
+			testFlowsMixedProtocols,
+			3,
+		},
+		{
+			"Filter for TCP on empty flows",
+			[]string{"tcp"},
+			[]*flowexporter.Connection{},
+			0,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			// Test the DumpFlows implementation of connTrackSystem
+			mockNetlinkCT := connectionstest.NewMockNetFilterConnTrack(ctrl)
+			connDumperDPSystem := NewConnTrackSystem(nodeConfig, svcCIDR, netip.Prefix{}, false, filter.NewProtocolFilter(tc.protocols))
 
-	connDumperDPSystem.connTrack = mockNetlinkCT
-	// Set expects for mocks
-	mockNetlinkCT.EXPECT().Dial().Return(nil)
-	mockNetlinkCT.EXPECT().DumpFlowsInCtZone(uint16(openflow.CtZone)).Return(testFlows, nil)
+			connDumperDPSystem.connTrack = mockNetlinkCT
+			// Set expects for mocks
+			mockNetlinkCT.EXPECT().Dial().Return(nil)
+			mockNetlinkCT.EXPECT().DumpFlowsInCtZone(uint16(openflow.CtZone)).Return(slices.Clone(tc.testFlows), nil)
 
-	conns, totalConns, err := connDumperDPSystem.DumpFlows(openflow.CtZone)
-	assert.NoErrorf(t, err, "Dump flows function returned error: %v", err)
-	assert.Equal(t, 1, len(conns), "number of filtered connections should be equal")
-	assert.Equal(t, len(testFlows), totalConns, "Number of connections in conntrack table should be equal to testFlows")
+			conns, totalConns, err := connDumperDPSystem.DumpFlows(openflow.CtZone)
+			assert.NoErrorf(t, err, "Dump flows function returned error: %v", err)
```

Comment:
shouldn't this be `require.NoError`?

Additionally, there is no need to explicitly log the error as it will be done automatically by the testify library.

Just realized that this code is not yours, but still a good opportunity to improve it